filament/src/driver/vulkan/VulkanHandles.cpp

```diff
@@ -154,17 +154,11 @@ VkExtent2D VulkanRenderTarget::getExtent() const {
 }
 
 VulkanAttachment VulkanRenderTarget::getColor() const {
-    if (mOffscreen) {
-        return mColor;
-    }
-    return getSwapContext(mContext).attachment;
+    return mOffscreen ? mColor : getSwapContext(mContext).attachment;
 }
 
 VulkanAttachment VulkanRenderTarget::getDepth() const {
-    if (mOffscreen) {
-        return mDepth;
-    }
-    return mContext.currentSurface->depth;
+    return mOffscreen ? mDepth : VulkanAttachment {};
```

Comment:
At some point we'll need to move this decision out of the driver. EGL has the same issue: our drivers "know" that the onscreen buffer doesn't need depth. Also, what is "mOffscreen" here? Not all of our off-screen buffers need depth (only the one used by the color pass, all the others currently don't need depth).

Reply:
Understood -- it should be easy to resurrect the depth buffer in the future when we do this.

If it is not a swap chain buffer, then mOffscreen is true, therefore VulkanRenderTarget owns the attachments, so we return mDepth.  Yes, mDepth is non-null only if the render target was created with TargetBufferFlags::DEPTH.

Reply:
SG thanks